softgroup/basis/common-router/common-router-api/src/main/java/com/softgroup/common/router/api/AbstractRouterHandler.java

```diff
@@ -1,18 +1,25 @@
 package com.softgroup.common.router.api;
 
+import com.softgroup.common.protocol.ProtocolBeansFactory;
 import com.softgroup.common.protocol.Request;
 import com.softgroup.common.protocol.Response;
+import com.softgroup.common.protocol.ResponseStatus;
+import com.softgroup.common.router.api.factory.Factory;
+import org.springframework.beans.factory.annotation.Autowired;
 
-public abstract class AbstractRouterHandler<T extends Handler> implements RouterHandler {
+public abstract class AbstractRouterHandler<T extends Factory> implements RouterHandler {
 
-	@Override
-	public String getRouteKey(Request<?> msg) {
-		return null;
-	}
-
-	@Override
-	public Response<?> handle(Request<?> msg) {
-		return null;
-	}
+    @Autowired
+    T factory;
 
+    @Override
+    public Response<?> handle(Request<?> msg) {
+        try {
+            return factory.getHandler(msg).handle(msg);
+        }catch (NullPointerException e){
```

Comment:
@melnyknikolay Don't use null pointer exception to check missing handler. Exceptions are slow and should not be used for cases when you can deal with exception situation. Here check if handler!=null.